franka-interface/src/skills/joint_torque_skill.cpp

```diff
@@ -0,0 +1,141 @@
+//
+// Created by Ruthrash Hari on 7/9/23
+//
+#include "franka-interface/skills/joint_torque_skill.h"
+#include <franka/robot.h>
+
+#include "franka-interface/robot_state_data.h"
+#include "franka-interface/run_loop.h"
+#include "franka-interface/run_loop_shared_memory_handler.h"
+#include "franka-interface/feedback_controller/set_internal_impedance_feedback_controller.h"
+#include "franka-interface/trajectory_generator/joint_trajectory_generator.h"
+#include <franka-interface-common/run_loop_process_info.h>
+
+////// unneccessary maybe?
+
+void JointTorqueSkill::limit_current_joint_torques(double period) {
+  for(int i = 0; i < 7; i++) {
+    if(std::abs(current_joint_torques_[i]) > max_joint_torques_[i] * safety_factor) {
+      if(current_joint_torques_[i] > 0) {
+        current_joint_torques_[i] = max_joint_torques_[i] * safety_factor;
+      } else {
+        current_joint_torques_[i] = -max_joint_torques_[i] * safety_factor;
+      }
+    }
+    current_joint_rotatums_[i] = (current_joint_torques_[i] - previous_joint_torques_[i]) / period;
+
+    if(std::abs(current_joint_rotatums_[i]) > max_joint_rotatums_[i] * safety_factor) {
+      if(current_joint_rotatums_[i] > 0) {
+        current_joint_rotatums_[i] = max_joint_rotatums_[i] * safety_factor;
+      } else {
+        current_joint_rotatums_[i] = -max_joint_rotatums_[i] * safety_factor;
+      }
+    }
+    current_joint_torques_[i] = previous_joint_torques_[i] + current_joint_rotatums_[i] * period;
```

Comment:
isn't this logic really discontinuous, i.e. at t = 0, we have 0 torques and then just directly at the next state we have a large torque value? 
Since this code seems to be doing `q_t  = q_{t-1} + dt * (q^{desired}_{t-1} - q_{t-1}) / dt` (basically combining 25 and 34). or am I missing something here?

Reply:
I think this is fine,
- at t = 0 we are applying 0 torque (libfranka applies torque to cancel gravity in backend) and at t = 1, we are limiting the commanded torque by limiting joint_rotatum to a fraction of the maximum in case it is greater than maximum joint_rotatum. This should limit discontinuity. 
- Please let me know if you think otherwise or if you think this could be done in a better way.